app/template/rest.py

```diff
@@ -211,9 +214,21 @@ def preview_letter_template_by_notification_id(service_id, notification_id, file
         content = base64.b64encode(pdf_file).decode('utf-8')
 
         if file_type == 'png':
-            url = '{}/precompiled-preview.png{}'.format(
-                current_app.config['TEMPLATE_PREVIEW_API_HOST'],
-                '?page={}'.format(page) if page else ''
+
+            try:
+                page_number = page if page else "0"
+                pdf_page = extract_page_from_pdf(BytesIO(pdf_file), int(page_number) - 1)
+                content = base64.b64encode(pdf_page).decode('utf-8')
+            except PdfReadError:
+                current_app.logger.exception(
```

Comment:
Do we need this `current_app.logger.exception` here, as the error handler in `errors.py` should already log exceptions.